src/btree/bt_walk.c

```diff
@@ -255,12 +255,18 @@ __tree_walk_internal(WT_SESSION_IMPL *session, WT_REF **refp, uint64_t *walkcntp
     restart_sleep = restart_yield = 0;
     empty_internal = false;
 
-    /* Ensure we have a snapshot to check visibility or we only check global visibility.
-     */
+    /* Ensure we have a snapshot to check visibility or we only check global visibility. */
     WT_ASSERT(session, LF_ISSET(WT_READ_VISIBLE_ALL) || F_ISSET(session->txn, WT_TXN_HAS_SNAPSHOT));
 
-    /* All current tree walks skip deleted pages. */
-    LF_SET(WT_READ_SKIP_DELETED);
+    /*
+     * We're not supposed to walk trees without root pages. As this has not always been the case,
+     * assert to debug that change.
+     */
+    WT_ASSERT(session, btree->root.page != NULL);
+
+    /* Check whether deleted pages can be skipped. */
+    if (!LF_ISSET(WT_READ_DELETED_SKIP))
+        LF_SET(WT_READ_DELETED_CHECK);
```

Comment:
I don't understand what this flag is for. Where do we actually check this flag to change the control flow?